src/main/java/net/sf/jabref/remote/DBConnector.java

```diff
@@ -0,0 +1,110 @@
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU General Public License for more details.
+
+    You should have received a copy of the GNU General Public License along
+    with this program; if not, write to the Free Software Foundation, Inc.,
+    51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+*/
+package net.sf.jabref.remote;
+
+import java.sql.Connection;
+import java.sql.DriverManager;
+import java.sql.SQLException;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+
+/**
+ * Used to establish connections between JabRef and database systems like MySQL, PostgreSQL and Oracle.
+ */
+public class DBConnector {
+
+    private static final Log LOGGER = LogFactory.getLog(DBConnector.class);
+
+
+    /**
+     * Determines the suitable driver and retrieves a working SQL Connection in normal case.
+     * A default port is going to be taken.
+     *
+     * @param dbType Enum entry of {@link DBType} which determines the driver
+     * @param host Hostname, Domain or IP address
+     * @param database An already existent database name.
+     * @param user Username
+     * @param password Password
+     * @return
+     * @throws ClassNotFoundException Thrown if no suitable drivers were found
+     * @throws SQLException Thrown if connection has failed
+     */
+    public static Connection getNewConnection(DBType dbType, String host, String database, String user, String password)
+            throws ClassNotFoundException, SQLException {
+        return getNewConnection(dbType, host, getDefaultPort(dbType), database, user, password);
+    }
+
+    /**
+     * Determines the suitable driver and retrieves a working SQL Connection in normal case.
+     *
+     * @param dbType Enum entry of {@link DBType} which determines the driver
+     * @param host Hostname, Domain or IP address
+     * @param port Port number the server is listening on
+     * @param database An already existent database name.
+     * @param user Username
+     * @param password Password
+     * @return
+     * @throws ClassNotFoundException Thrown if no suitable drivers were found
+     * @throws SQLException Thrown if connection has failed
+     */
+    public static Connection getNewConnection(DBType dbType, String host, int port, String database, String user,
+            String password) throws ClassNotFoundException, SQLException {
+
+        String url = "jdbc:";
+
+        try {
+            if (dbType == DBType.MYSQL) {
+                Class.forName("com.mysql.jdbc.Driver");
+                url = url + "mysql://" + host + ":" + port + "/" + database;
+            } else if (dbType == DBType.ORACLE) {
+                Class.forName("oracle.jdbc.driver.OracleDriver");
+                url = url + "oracle:thin:@" + host + ":" + port + ":" + database;
+            } else if (dbType == DBType.POSTGRESQL) {
+                Class.forName("org.postgresql.Driver");
```

Comment:
The calls initialize the driver: At least for mysql necessary
https://dev.mysql.com/doc/connector-j/5.1/en/connector-j-usagenotes-connect-drivermanager.html